tests/test_selection/test_drop_high_psi_features.py

```diff
@@ -0,0 +1,543 @@
+    df = pd.DataFrame(
+        {
+            "time": [date(2012, 6, it) for it in range(1, 31)],
+            "A": [it for it in range(0, 30)],
+            "B": [1, 2, 3, 4, 5, 6] * 5,
+        }
+    )
+
+    expected_psi = {"A": 7.745116608833365, "B": 0.1621860432432658}
+
+    test = DropHighPSIFeatures(split_frac=0.5, split_distinct=True)
+    test.fit_transform(df)
+
+    assert test.cut_off_ == 14.5
+    assert test.psi_values_ == pytest.approx(expected_psi)
+
+
+type_test = [
+    ("A", 14, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]),
+    ("B", 1, [0, 3, 4, 7, 8, 11, 12, 15, 16, 19]),
+    ("C", ["A"], [0, 4, 8, 12, 16]),
+    ("time", date(2019, 1, 4), [0, 1, 2, 3]),
+]
+
+
+@pytest.mark.parametrize("col, cut_off, expected", type_test)
+def test_split_using_cut_off(col, cut_off, expected, df_mixed_types):
+    """Test the cut off for different data types."""
+    test = DropHighPSIFeatures(split_col=col, cut_off=cut_off)
+    a, b = test._split_dataframe(df_mixed_types)
+
+    pd.testing.assert_frame_equal(a, df_mixed_types.loc[expected])
+    pd.testing.assert_frame_equal(
+        b, df_mixed_types.loc[~df_mixed_types.index.isin(expected)]
+    )
+
+
+split_distinct_test = [
+    ("A", [number for number in range(0, 100)]),
+    (
+        "B",
+        (
+            [0, 1, 2, 10, 11, 12, 20, 21, 22, 30, 31, 32, 40]
+            + [41, 42, 50, 51, 52, 60, 61, 62, 70, 71, 72, 80, 81]
+            + [82, 90, 91, 92, 100, 101, 102, 110, 111, 112, 120, 121, 122]
+            + [130, 131, 132, 140, 141, 142, 150, 151, 152, 160, 161, 162, 170]
+            + [171, 172, 180, 181, 182, 190, 191, 192]
+        ),
+    ),
+    (
+        "C",
+        (
+            [0, 1, 2, 10, 11, 12, 20, 21, 22, 30, 31, 32, 40]
+            + [41, 42, 50, 51, 52, 60, 61, 62, 70, 71, 72, 80, 81]
+            + [82, 90, 91, 92, 100, 101, 102, 110, 111, 112, 120, 121, 122]
+            + [130, 131, 132, 140, 141, 142, 150, 151, 152, 160, 161, 162, 170]
+            + [171, 172, 180, 181, 182, 190, 191, 192]
+        ),
+    ),
+    (
+        "time",
+        (
+            [0, 1, 2, 5, 6, 7, 10, 11, 12, 15, 16, 17, 20, 21, 22, 25, 26]
+            + [27, 30, 31, 32, 35, 36, 37, 40, 41, 42, 45, 46, 47, 50, 51, 52, 55]
+            + [56, 57, 60, 61, 62, 65, 66, 67, 70, 71, 72, 75, 76, 77, 80, 81, 82]
+            + [85, 86, 87, 90, 91, 92, 95, 96, 97]
+        ),
+    ),
+]
+
+
+@pytest.mark.parametrize("col, expected_index", split_distinct_test)
+def test_split_distinct(col, expected_index):
+    """Test the cut off for different data types.
+
+    For columns B, C and time we have 6 distinct values, 5 appearing 20 times and
+    1 appearing 100 times. A 50% split based on the number of values will results
+    in 2 groups of 3. One has 60 appearances (in total) and the other has 140.
+    """
+    data = pd.DataFrame(
+        {
+            "A": [it for it in range(0, 200)],
+            "B": [1, 2, 3, 4, 5, 6, 6, 6, 6, 6] * 20,
```

Comment:
If B was instead [1, 1, 1, 1, 1, 2, 3, 4, 5, 6], does this still work as expected? 

could we add a test for this example? 

because I can't see it happening as expected from the logic in the class.